SmartAppointmentSystem.Business/Implementations/TokenService.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.IdentityModel.Tokens;
+using SmartAppointmentSystem.Business.Contracts;
+using SmartAppointmentSystem.Business.DTOs;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+namespace SmartAppointmentSystem.Business.Implementations;
+
+public class TokenService : ITokenService
+{
+    public Task<GenerateTokenResponseDTO> GenerateToken(GenerateTokenRequestDTO request)
+    {
+        var secretKey = Environment.GetEnvironmentVariable("AppSettings__Secret");
+        if (string.IsNullOrEmpty(secretKey))
+        {
+            throw new Exception("JWT Secret Key not found.");
+        }
+        SymmetricSecurityKey symmetricSecurityKey = new SymmetricSecurityKey(Encoding.ASCII.GetBytes(secretKey));
+        var dateTimeNow = DateTime.UtcNow;
+
+        JwtSecurityToken jwt = new JwtSecurityToken(
+            claims: new List<Claim>
+            {
+                new Claim("UserId", request.UserId.ToString()),
+                new Claim("Name", request.Name),
+                new Claim("Mail", request.Mail),
+                new Claim(ClaimTypes.Role, request.Role)
+            },
+            notBefore: dateTimeNow,
+            expires: dateTimeNow.AddMinutes(300),
```

Comment:
300 is meaningless here. Move this value to appsettings.json and read 300 value from appsettings.